src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -683,13 +697,9 @@ private boolean isCommentInBeginningOfLine(DetailAST comment) {
      * @return true, if token is preceded by javadoc comment.
      */
     private static boolean isPrecededByJavadoc(DetailAST token) {
-        boolean result = false;
         final DetailAST previous = token.getPreviousSibling();
-        if (previous.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
-                && JavadocUtil.isJavadocComment(previous.getFirstChild().getText())) {
-            result = true;
-        }
-        return result;
+        return previous.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
+                && JavadocUtil.isJavadocComment(previous.getFirstChild().getText());
```

Comment:
please never mix functional fix in refactoring in same commit.

I would do:
```
result = previous.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
                && JavadocUtil.isJavadocComment(previous.getFirstChild().getText());
return result;
```
as it would ease debugging, and will easily be optimized by compiler. I am old-school.

Reply:
@romani there are plenty of methods like this (return expression without unnecessary variable), even in this class almost all other methods do it in same way, e.g. this one (2 lines below current)
```
private static boolean isComment(DetailAST ast) {
        return ast.getType() == TokenTypes.SINGLE_LINE_COMMENT
                   || ast.getType() == TokenTypes.BLOCK_COMMENT_BEGIN;
    }
```

What is wrong with debugging?

Reply:
1) I do not like long and complicated expressions.
2) I like single exit point concept in methods (it force to organize code well, we benefited a lot after enforcement, even process was also not welcomed at  the beginning), when you have code like this you put a break point to `return result` and you easily have final value, if parts of expressions do not do side effects, you can re-evaluate parts of expression to get to know which part of it result in unexpected result.
3) simple expressions are growing in more complicated , and I not always have ability to catch it during review. I am not ready to design Check on this, as there are a lot of nuances in objective term `complicated expression`.

Lets vote on this.
@rnveach  @pbludov , please vote.

Reply:
1. how come that such expression became complicated? I can agree if there is some non-obvious things and variable name can clarify it, but create useless `result` variable is not the case.
2. I do not see any conflicts on this. You can evaluate expression if you put breakpoint. Conditional breakpoints solve this problem.

Reply:
I'm agree with @strkkk. The variable `result` is redundant here.
Now the code is strictly sequential and more readable (for me).
Variables are needed when there are *real* branches.

Reply:
> how come that such expression became complicated?

in future, there will be some additional request for checking in this condition and next contributor will simply extend expression with extra code

> You can evaluate expression if you put breakpoint.

and now please count a time, you put break point and now you try to accurately select whole expression in IDE to evaluate it. Compare it with simply hovering it over "result" variable. 

> Now the code is strictly sequential and more readable (for me).

with extra variable it it also sequential.

> Variables are needed when there are real branches.

I do not understand this. Variables help to organize code and keep some naming context.

Reply:
1. We can make some limit (if there are several conditions, it should be extracted to variables) but it will definitely more than 2, as we have here.
2. I don't see major issue in it.

Reply:
> 1.

 Limit will not work, as there is required to detect "the same" pattern between parts of expression. It is a bit more complicated, I am not ready to provide all rules for such future Check. Expression with 2 items is not good example to discuss over. I just try to warn you that I am not ok to refactor whole project to make it single statement "return" with whole logic inlined.

> 2 

All Checkstyle violations are not a major issue. We focus on issues that economize minutes and seconds of user during reading, and even during debug (it is also a phase where engineers spend a lot of time). Where are areas where we still can improve(even in seconds, less actions, less distractions, less scrolling) reading experience of code. I am ok to inline variable definition by expression of IF, but I still see convenience to keep SINGLE exit point as "single" variable return (but I agree there are simple cases).